kubetest2-gke/deployer/commandutils.go

```diff
@@ -116,3 +117,36 @@ func getClusterCredentials(project, loc, cluster string) error {
 
 	return nil
 }
+
+// Resolve the current latest version in the given release channel.
+func resolveLatestVersionInChannel(loc, channelName string) (string, error) {
+	// Get the server config for the current location.
+	out, err := exec.Output(exec.RawCommand(
+		fmt.Sprintf("gcloud container get-server-config --format=\"yaml(channels:format='yaml(channel,validVersions)')\" %s", loc)))
+	if err != nil {
+		return "", fmt.Errorf("failed to get the server config: %w", err)
+	}
+
+	type Channel struct {
+		Name          string   `yaml:"channel"`
+		ValidVersions []string `yaml:"validVersions"`
+	}
+	type Channels struct {
+		Channels []Channel `yaml:"channels"`
+	}
+	var cs Channels
+	if err = yaml.Unmarshal(out, &cs); err != nil {
+		return "", fmt.Errorf("failed to unmarshal the server config: %w", err)
+	}
+
+	for _, channel := range cs.Channels {
+		if strings.EqualFold(channel.Name, channelName) {
+			if len(channel.ValidVersions) == 0 {
+				return "", fmt.Errorf("no valid versions for channel %q", channelName)
+			}
+			return channel.ValidVersions[0], nil
```

Comment:
are the returned values already sorted in descending order?

Reply:
Yes, they are already sorted in descending order.